src/Compilers/Test/Core/TargetFrameworkUtil.cs

```diff
@@ -322,6 +324,11 @@ private static ImmutableArray<PortableExecutableReference> LoadDynamicReferences
 
                 var type = assembly.GetType(assemblyName, throwOnError: true);
                 var prop = type.GetProperty("All", BindingFlags.Public | BindingFlags.Static);
+                if (prop is null)
+                {
+                    type = assembly.GetType(assemblyName + "+References", throwOnError: true);
+                    prop = type.GetProperty("All", BindingFlags.Public | BindingFlags.Static);
+                }
```

Comment:
What does this change represent? (To be honest, it's not obvious to me what `LoadDynamicReferences()` is for.)

Reply:
> What does this change represent? (To be honest, it's not obvious to me what LoadDynamicReferences() is for.)

@jaredpar? This part is inherited from #68183.

Reply:
For every target framework that we test against we have effectively an assembly that encapsulates the reference assemblies. So for `net7.0` we have `Basic.Reference.Assemblies.Net70`. These are each about 5MB. On its' own that is not too bad. The problem is that any `<PackageReference>` in this project then they will get copied to every single unit test project in our solution. 

We have ~60 unit test projects many of which have multiple target framework. Taken together a reference added here will appear lets say ~80 times in our artifacts directory. Yet in reality only the compiler tests actually use most of these references. The IDE uses a sparing number of them. The `LoadDynamicReference` method gives us the API we want from testing `TargetFramework.Net70` but the actual dll deployment / load is only incurred by the project that uses it. It's much more space efficient. 

The reason for the change is the latest version of the ref assemblies changed the type layout a bit. Had to account for that. 

Note: please let me know if the doc comment on `LoadDynamicReference` isn't sufficiently explaining this. Will update.